var/spack/repos/builtin/packages/netcdf-fortran/package.py

```diff
@@ -0,0 +1,18 @@
+from spack import *
+
+class NetcdfFortran(Package):
+    """Fortran interface for NetCDF4"""
+
+    homepage = "http://www.unidata.ucar.edu/downloads/netcdf/netcdf-cxx/index.jsp"
+    url      = "http://www.unidata.ucar.edu/downloads/netcdf/ftp/netcdf-fortran-4.4.3.tar.gz"
+
+    version('4.4.3', 'bfd4ae23a34635b273d3eb0d91cbde9e')
+
+    variant('mpi', default=True, description='Enables MPI parallelism')
```

Comment:
It would be nice if the Fortran interface just figured out that it should use MPI based on how netcdf is built.  `netcdf-fortran ^netcdf~mpi`  and `netcdf-fortran ^netcdf+mpi`  are two different things, so dependents would be able to get the right behavior if they just did this:

```
depends_on('netcdf-fortran')
depends_on('netcdf+mpi')
```

The constraint on `netcdf` will get passed through and merged with any from other packages or the command line.  You could argue that this is simpler and easier for packagers:

```
depends_on('netcdf-fortran+mpi')
```

It's a matter of whether you want to maintain the variants only in `netcdf` or also in the bindings.  Either way, the variant in `netcdf-fortran` should either do something or be removed.  If it's removed, you can check whether `netcdf` was built with MPI like this:

```
if 'mpi' in spec['netcdf']:
    # Add some configure args to support MPI.
```

As usual, `spec['mpi']` will get you the `mpi` dependency, but I would recommend checking for `mpi` in `spec['netcdf']` to make sure that it's actually *netcdf* that was built with MPI, and not some other indirect dep.  Seems unlikely for this particular package, so maybe that's overly cautious, but who knows.

Once you know fi `netcdf` is built with MPI, you can add the necessary config args to the fortran bindings.  I would assume that the fortran bindings need to know what MPI they're using and it's good to be explicit with the arguments so that you don't inadvertently bring in a system MPI.